### PR TITLE
Send DM when divorce

### PR DESCRIPTION
**Pull request by iiAmFilippi**
https://github.com/LorittaBot/Loritta/issues/2289

### CODE REVIEW

**Comment by SrGaabriel** [comment]
This will throw exceptions if the user has its DM closed or blocked Loritta.

**Comment by SrGaabriel** [comment]
And you should fix the code formatting.

**Comment by iiAmFilippi** [comment]
> 
> 
> And you should fix the code formatting.

as well?

**Comment by MrPowerGamerBR** [comment]
I think that it would be better if the message was prettier in a embed:

Something like this:
image

**Comment by iiAmFilippi** [comment]
already made the changes

File: loritta-discord/src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/social/DivorceCommand.kt [hunk 1]
```diff
@@ -55,6 +60,19 @@ class DivorceCommand : AbstractCommand("divorce", listOf("divorciar"), CommandCa
                                     locale["$LOCALE_PREFIX.divorced", Emotes.LORI_HUG]
                             )
 					)
+					
+					try {
+						val userDM = user?.openPrivateChannel()?.await()
```
Comment (SrGaabriel) [inline loritta-discord/src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/social/DivorceCommand.kt:65]: Try using this and then remove the null safety characters:

```suggestion
						val userPrivateChannel = user?.openPrivateChannel()?.await() ?: return
```

File: loritta-discord/src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/social/DivorceCommand.kt [hunk 2]
```diff
@@ -55,6 +60,19 @@ class DivorceCommand : AbstractCommand("divorce", listOf("divorciar"), CommandCa
                                     locale["$LOCALE_PREFIX.divorced", Emotes.LORI_HUG]
                             )
 					)
```
Comment (SrGaabriel) [inline loritta-discord/src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/social/DivorceCommand.kt:62]: I think you should fix the code formatting, like on this line.

File: loritta-discord/src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/social/DivorceCommand.kt [hunk 3]
```diff
@@ -55,6 +60,19 @@ class DivorceCommand : AbstractCommand("divorce", listOf("divorciar"), CommandCa
...
+					
+					try {
+						val userDM = user?.openPrivateChannel()?.await()
+
+						userDM?.sendMessage(
+							EmbedBuilder()
+								.setTitle(locale["$LOCALE_PREFIX.divorcedTitle"])
+								.setDescription(locale["$LOCALE_PREFIX.divorcedDescription", context.userHandle.name])
+								.setThumbnail("https://cdn.discordapp.com/emojis/556524143281963008.png?size=2048")
```
Comment (SrGaabriel) [inline loritta-discord/src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/social/DivorceCommand.kt:71]: Migrate this URI to the companion object

File: loritta-discord/src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/social/DivorceCommand.kt [hunk 4]
```diff
@@ -25,6 +28,8 @@ class DivorceCommand : AbstractCommand("divorce", listOf("divorciar"), CommandCa
 
 	override suspend fun run(context: CommandContext,locale: BaseLocale) {
 		val marriage = loritta.newSuspendedTransaction { context.lorittaUser.profile.marriage }
+		val userId = if (context.userHandle.idLong == marriage?.user2) marriage.user1 else marriage?.user2
+		val user = lorittaShards.getUserById(userId)
```
Comment (SrGaabriel) [inline loritta-discord/src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/social/DivorceCommand.kt:32]: You should get the marriage data using the `ProfileUtils.kt` class.

**Review by SrGaabriel — CHANGES_REQUESTED** [review]

File: loritta-discord/src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/social/DivorceCommand.kt (hunk 3 above)
Comment (MrPowerGamerBR) [reply loritta-discord/src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/social/DivorceCommand.kt:71]: ^ That's not really necessary but would be nice to change it

**Review by MrPowerGamerBR — CHANGES_REQUESTED** [review]

**Comment by MrPowerGamerBR** [comment]
pls fix the stuff that Gabriel suggested

**Comment by iiAmFilippi** [comment]
Done.

**Comment by MrPowerGamerBR** [comment]
looks good to me

**Comment by iiAmFilippi** [comment]
:lori_ameno:

**Pull request merged**